### PR TITLE
fix: always route SMS to throttled queue when sender is set

### PR DESCRIPTION
**Pull request by AntoineAugusti**
## Current behaviour
- I have a service with a custom SMS sender
- I upload a spreadsheet sending more than 200 SMS
- Those SMS are routed to the usual SMS queue and are sent faster than 1 SMS per second

## Expected behaviour
SMS are throttled at 1 SMS per second no matter what, to respect telecommunication rate limits.

## Description

Make sure that SMS with a custom sender are always routed to the throttled SMS SQS queue, even if another queue was requested.

This makes sure that we don't go over rate limits for each long code number.

We had a problem in production because a client sent a large number of SMS through a spreadsheet and they were routed to the bulk queue thanks to the recent work done in https://github.com/cds-snc/notification-api/pull/1296.

### CODE REVIEW

File: tests/app/celery/test_tasks.py [hunk 1]
```diff
@@ -584,6 +584,32 @@ def test_should_route_save_sms_task_to_bulk_on_large_csv_file(notify_db, notify_
     assert mocked_deliver_sms.called
 
 
+def test_should_route_save_sms_task_to_throttled_queue_on_large_csv_file_if_custom_sms_sender(
+    notify_db, notify_db_session, mocker
+):
+    service = create_service_with_defined_sms_sender(sms_sender_value="3433061234")
```
Comment (AntoineAugusti) [inline tests/app/celery/test_tasks.py:590]: Setting a default SMS sender with a custom number

File: tests/app/celery/test_tasks.py [hunk 2]
```diff
@@ -584,6 +584,32 @@ def test_should_route_save_sms_task_to_bulk_on_large_csv_file(notify_db, notify_
...
+        notification_id,
+        encryption.encrypt(notification),
+    )
+
+    persisted_notification = Notification.query.one()
+    provider_tasks.deliver_throttled_sms.apply_async.assert_called_once_with(
+        [str(persisted_notification.id)], queue="send-throttled-sms-tasks"
+    )
+    mocked_deliver_sms.assert_not_called()
+    mocked_deliver_throttled_sms.assert_called_once()
```
Comment (AntoineAugusti) [inline tests/app/celery/test_tasks.py:610]: Making sure this is routed to the appropriate queue

File: tests/app/notifications/test_process_notification.py [hunk 3]
```diff
@@ -289,7 +289,7 @@ def test_persist_notification_increments_cache_if_key_exists(sample_template, sa
             "sms",
             "normal",
             "+14383898585",
-            "research-mode-tasks",
+            "send-throttled-sms-tasks",
```
Comment (AntoineAugusti) [inline tests/app/notifications/test_process_notification.py:292]: Does not care that this is research mode and routed to the throttled queue

File: tests/app/notifications/test_process_notification.py [hunk 4]
```diff
@@ -356,7 +356,7 @@ def test_persist_notification_increments_cache_if_key_exists(sample_template, sa
             "sms",
             "normal",
             "+14383898585",
-            "notify-internal-tasks",
+            "send-throttled-sms-tasks",
```
Comment (AntoineAugusti) [inline tests/app/notifications/test_process_notification.py:359]: Does not care that a custom queue was requested, still routed to the throttled queue

File: app/notifications/process_notifications.py [hunk 5]
```diff
@@ -125,8 +125,7 @@ def send_notification_to_queue(notification, research_mode, queue=None):
         deliver_task = provider_tasks.deliver_sms
         if notification.sends_with_custom_number():
             deliver_task = provider_tasks.deliver_throttled_sms
-            if not queue:
-                queue = QueueNames.SEND_THROTTLED_SMS
+            queue = QueueNames.SEND_THROTTLED_SMS
```
Comment (AntoineAugusti) [inline app/notifications/process_notifications.py:128]: Removing the `if` is the actual fix

**Review by maxneuvians — APPROVED** [review]
LGTM - nice work!

**Pull request merged**